iModelJsNodeAddon/presentation/ECPresentationSerializer.cpp

```diff
@@ -1456,12 +1466,15 @@ std::unique_ptr<BoundQueryValue> IModelJsBoundQueryValueSerializer::_FromJson(Be
         }
     if (0 == strcmp("ValueSet", type))
         {
+        bvector<ECValue> ecValues;
         Utf8String valueType = json["valueType"].asCString();
         if (valueType.empty())
             return std::make_unique<BoundECValueSet>(bvector<ECValue>());
-        rapidjson::Document doc;
-        doc.Parse(json["value"].Stringify().c_str());
-        return std::make_unique<BoundRapidJsonValueSet>(doc, (PrimitiveType)ParsePrimitiveType(valueType));
+
+        rapidjson::Document values;
+        values.Parse(json["value"].Stringify().c_str());
+        return std::make_unique<BoundECValueSet>(GetECValueSetFromJson((PrimitiveType)ParsePrimitiveType(valueType), values));

```

Comment:
let's check if `ParsePrimitiveType(valueType)` returns non zero result. return empty set if it does